scipy/conftest.py

```diff
@@ -236,3 +236,133 @@ def skip_if_array_api(xp, request):
+
+    # scipy.stats deliberately emits UserWarnings sometimes
+    user_w = ['scipy.stats.anderson_ksamp', 'scipy.stats.kurtosistest',
+              'scipy.stats.normaltest', 'scipy.sparse.linalg.norm']
+    for name in user_w:
+        known_warnings[name] = dict(category=UserWarning)
+
+    # additional one-off warnings to filter
+    dct = {
+        'scipy.sparse.linalg.norm':
+            dict(category=UserWarning, message="Exited at iteration"),
+        # tutorials
+        'linalg.rst':
+            dict(message='the matrix subclass is not',
+                 category=PendingDeprecationWarning),
+        'stats.rst':
+            dict(message='The maximum number of subdivisions',
+                 category=integrate.IntegrationWarning),
+    }
+    known_warnings.update(dct)
+
+    # these legitimately emit warnings in examples
+    from scipy.signal._filter_design import BadCoefficients
+    legit = set('scipy.signal.normalize')
+
+    # Now, the meat of the matter: filter warnings,
+    # also control the random seed for each doctest.
+
+    # XXX: this matches the refguide-check behavior, but is a tad strange:
+    # makes sure that the seed the old-fashioned np.random* methods is *NOT*
+    # reproducible but the new-style `default_rng()` *IS* repoducible.
+    # Should these two be either both repro or both not repro?
+
+    from scipy._lib._util import _fixed_default_rng
+    import numpy as np
+    with _fixed_default_rng():
+        np.random.seed(None)
+        with warnings.catch_warnings():
+            if test.name in known_warnings:
+                warnings.filterwarnings('ignore', **known_warnings[test.name])
+                yield
+            elif test.name in legit:
+                yield
+            else:
+                warnings.simplefilter('error', Warning)
+                yield
+
+
+dt_config.user_context_mgr = warnings_errors_and_rng
+dt_config.skiplist = set([
+    'scipy.linalg.LinAlgError',     # comes from numpy
+    'scipy.fftpack.fftshift',       # fftpack stuff is also from numpy
+    'scipy.fftpack.ifftshift',
+    'scipy.fftpack.fftfreq',
+    'scipy.special.sinc',           # sinc is from numpy
+    'scipy.optimize.show_options',  # does not have much to doctest
+    'scipy.signal.normalize',       # manipulates warnings (XXX temp skip)
+    'scipy.sparse.linalg.norm',     # XXX temp skip
+])
+
+# help pytest collection a bit: these names are either private (distributions),
+# or just do not need doctesting.
+dt_config.pytest_extra_ignore = [
+    "scipy.stats.distributions",
+    "scipy.optimize.cython_optimize",
+    "scipy.test",
+    "scipy.show_config",
+]
+
+dt_config.pytest_extra_xfail = {
+    # name: reason
+    "io.rst": "",
+    "ND_regular_grid.rst": "ReST parser limitation",
+    "ND_unstructured.rst": "matplotlib deprecation",
+    "extrapolation_examples.rst": "ReST parser limitation",
+    "sampling_pinv.rst": "__cinit__ unexpected argument",
```

Comment:
@tirthasheshpatel had to xfail the whole tutorial, looks like the code has changed after the docs were written?